src/nvim/eval/typval.h

```diff
@@ -413,11 +413,17 @@ static inline void list_log(const list_T *const l,
 #define TV_DICT_HI2DI(hi) \
     ((dictitem_T *)((hi)->hi_key - offsetof(dictitem_T, di_key)))
 
+static inline void tv_list_ref(list_T *const l)
+  REAL_FATTR_ALWAYS_INLINE;
+
+static inline void tv_list_set_ret(typval_T *const tv, list_T *const l)
+  REAL_FATTR_ALWAYS_INLINE REAL_FATTR_NONNULL_ARG(1);
```

Comment:
This should be just before the definition: it is already easy to forget that declaration exists when updating definition given that `*.c` files conveniently have no declarations, moving it elsewhere makes that even easier.